tests/unittest/FaceliftObserverTest.cpp

```diff
@@ -0,0 +1,273 @@
+    // Observer to execute the callback type "callOnceReady"
+    SingleTimeObserver< std::function<void()> > obs2 { std::bind(&Counter::incValue, &c2) };
+    // Observer with nullptr
+    StandartObserver < std::function<void()> > obs3 { nullptr };
+
+    // The "Proxy" is the source of the signal of interest
+    IPCProxy<InterfaceBase> proxy { nullptr };
+    // The "IsReadyObserver" is the watcher of the signal of interest in "Proxy" and contain observers to execute: "callOnReady", "callOnceReady"
+    IsReadyObserver readyObserver{};
+
+    ~IPCProxyTest() {}
+};
+
+TEST_F(IPCProxyTest, proxyReadyIsFalse)
+{
+    /**
+     * @brief checking if proxy is not ready and signal exists, observer should not be executed.
+     * @param set to Proxy in state "false"
+     * @param set 2 observers
+     */
+    // Set state ready() of Proxy to disable
+    proxy.setReadyTest(false);
+
+    // Setting up an "IsReadyObserver" to watch out for signal in "Proxy"
+    readyObserver.watch(&proxy, &IPCProxy<InterfaceBase>::readyChanged, &IPCProxy<InterfaceBase>::readyTest);
+    // Set observers to "IsReadyObserver"
+    const auto expected = QVector< QPointer<IObserver> >{ &obs1, &obs2 };
+    readyObserver.setObservers(expected);
+
+    // Check values before calling a signals
+    ASSERT_EQ(c1.getValue(), 0); // for StandartObserver ("callOnReady")
+    ASSERT_EQ(c2.getValue(), 0); // for SingleTimeObserver ("callOnceReady")
+
+    // Set signal tracking for "IsReadyObserver"
+    QSignalSpy spyObserver(&readyObserver, &IsReadyObserver::readyChanged);
+    ASSERT_EQ(spyObserver.isValid(), true);
+    spyObserver.clear();
+
+    // Set signal tracking for "Proxy"
+    QSignalSpy spyProxy(&proxy, &IPCProxy<InterfaceBase>::readyChanged );
+    ASSERT_EQ(spyProxy.isValid(), true);
+    spyProxy.clear();
+
+    // Check that the signals on "IsReadyObserver" and "Proxy" are absent
+    ASSERT_EQ(spyObserver.count(), countOfReceivedSignalsBeforeEmit);
+    ASSERT_EQ(spyProxy.count(), countOfReceivedSignalsBeforeEmit);
+
+    // Generate a some signals on the "Proxy" which is tracking on "IsReadyObserver"
+    for (auto i=0; i<countOfEmittedProxySignals; ++i) {
+        proxy.readyChanged();
+    }
+
+    // Check the number of received signals on "Proxy"
+    ASSERT_EQ(spyProxy.count(), countOfEmittedProxySignals);
+    // Check the number of received signals on "IsReadyObserver"
+    ASSERT_EQ(spyObserver.count(), countOfReceivedSignalsWhenReadyFalse);
+
+    // Check values after signals call
+    ASSERT_EQ(c1.getValue(), 0); // for StandartObserver ("callOnReady")
+    ASSERT_EQ(c2.getValue(), 0); // for SingleTimeObserver ("callOnceReady")
+}
+TEST_F(IPCProxyTest, proxyReadyIsTrue)
+{
+    /**
+     * @brief checking if proxy is ready and signal exists, all observer should be executed.
+     * @param set to Proxy in state "true"
+     * @param set 2 observers
+     */
```

Comment:
this is not a place for doxygen documentation!!!!

Reply:
Done